### ISSUE TITLE
[BUG] Half empty radius accounting messages after calling drop_accounting()

### ISSUE THREAD

**Issue by tijmenNL** [OpenSIPS/opensips#3028]
**OpenSIPS version you are running**

```
version: opensips 3.2.11 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 912530b5a
main.c compiled on  with gcc 8
```

**Describe the bug**
After updating to the latest 3.2 (from 3.2.9 - git revision: 4a04a661b), calling drop_accounting()  is behaving different and a radius request is still being transmitted with half of the data, it looks like (from/to tag and call id manually removed):

```
Tue Mar  7 14:18:03 2023
	Acct-Status-Type = Failed
	Service-Type = Sip-Session
	Sip-Response-Code = 403
	Sip-Method = Invite
	Event-Timestamp = 1678195083
	Sip-From-Tag = "..."
	Sip-To-Tag = "..."
	Acct-Session-Id = "..."
	X-Ascend-Dial-Number = "d\0079\213"
	X-Ascend-Route-IP = Route-IP-No
	NAS-Port = 5060
	Acct-Delay-Time = 0
	NAS-IP-Address = 127.0.0.1
	Proxy-State = 0x323030
	Acct-Unique-Session-Id = "..."
	Timestamp = 1678195083
	Request-Authenticator = Verified
```

This is causing half empty  accounting records.

**Expected behavior**

After drop_accounting() no request should be sent, just like it was before updating.

**OS/environment information**
 - Operating System: Debian 10.12

**Comment by tijmenNL** [comment]
Perhaps this is caused by: e2b698a7c6e71e1d272b8f7e330266427c86fe52

**Comment by liviuchircu** [comment]
Thank you for the report, @tijmenNL!  What I tried to fix in e2b698a was the fact that `drop_accounting()` was simply _not working_ with `"db"` accounting, for example with `db_flatstore` storage.  The so-called fix was seamlessly straight-forward (just a missing flag mask) and got the job done for my test case.

Let me see what went wrong, though, and how it could have led to this strange effect with RADIUS acc!

**Comment by liviuchircu** [comment]
Hey, @tijmenNL!  I know it's been a while here, but today we realized there was a regression in e2b698a, causing "too much" `drop_accounting()` in some specific scenarios (e.g., the whole accounting was dropped in cases where only `failed` accounting should have been dropped, for example).

Now, regarding your issue where you noticed additional records after merging in the commit, I still cannot explain it.  But, if it's still an issue and you can reproduce it, please re-open here so we can work on fixing it.  Cheers!

**Comment by tijmenNL** [comment]
Hi @liviuchircu ! I tested the patch and it fixes the problem!

I tested the 3.2 branch before and after the last 2 ACC patches and I see the difference in behaviour. Before it still goes wrong, after it seems to be correct.

To give some background, the scenario where it would output the half of the packet is the following:

There are 2 OpenSIPS instances on two machines. The caller is on machine 1, called is on machine 2.

Machine 1 is forwarding the call to 2 and 2 should do the accounting for the entire call. So 1 calls `drop_accounting()` then it forwards. In short the logic on machine 1 is:

```
$du = $avp(machine2);

resetflag("ACCOUNTING_FLAG");
drop_accounting();
record_route();

t_relay();
exit;
```

After this the call gets canceled and I see two radius packets, one from each machine. The one from machine 1 is an incomplete failed packet, while the one from 2 is correct. 

The cancel gets relayed to 2 without alteration (`t_relay(0x08);`) and no do_accounting gets called.

So my guess is that perhaps drop_accounting also did not drop the entire accounting before the fix, since the failed packet sort of half worked? It should not have sent the ACC failed at all, since `drop_accounting()` was called.

Anyway, thanks for the fix!
